text/0018-publish-update.md

```diff
@@ -0,0 +1,180 @@
+  publishing a dataset always makes all previous history entries available.
+
+  By default publish also publishes your dataset to all configured registries,
+  to publish to only specific registries, use the --
+
+Examples:
+  # publish the latest version of a dataset
+  $ qri publish me/dataset
+
+  # publish a dataset (and all previous versions) at a specific version
+  $ qri publish me/dataset@/ipfs/QmFoo...
+
+  # unpublish a dataset
+  $ qri publish -d me/dataset
+
+Flags:
+  --delete -d       unpublish the specified dataset
+  --no-registry -r  don't publish to the registry
+  --registries      comma separated list of registries to publish to, overrides --no-registry
+```
+
+#### API:
+```
+# publish a reference:
+POST http://localhost:2505/publish/me/dataset
+
+# publish a dataset (and all previous versions) at a specific version
+POST http://localhost:2505/publish/me/dataset/at/ipfs/QmFoo...
+
+# unpublish a dataset
+DELETE http://localhost:2505/publish/me/dataset
+```
+
+## Update:
+#### CLI:
+```
+$ qri update --help
+
+Description:
+  Update fast-forwards your dataset to the latest known version. If the dataset
+  is owned by a peer, update will ask the peer for any new versions and download
+  them.
+
+  Calling update on a dataset you own will advance your dataset by re-running 
+  any specified transform, creating a new version of your dataset in the process.
+  If your dataset doesn't have a transform script, update does nothing.
+
+Examples:
+  # get the freshest version of a dataset from a peer
+  qri update other_person/dataset
+
+  # update your local dataset by re-running the dataset transform
+  qri update me/dataset_with_transform
+
+  # supply secrets to an update, publish on successful run
+  qri update me/dataset_with_transform -p --secrets=keyboard,cat
+
+Flags:
+  --publish, -p     publish successful updates
+  --title, -t       supply a 
+  --message, -m     supply a commit message for local updates
+  --secrets, -s     specify transform secrets 
+```
+
+### API:
+```
+# get the freshest version of a dataset from a peer
+POST http://localhost:2503/update/other_person/dataset
+
+# update your local dataset by re-running the dataset transform
+POST http://localhost:2503/update/me/dataset_with_transform
+
+# supply secrets to an update, publish on successful run
+POST http://localhost:2503/update/me/dataset_with_transform?publish=true -H secrets=keyboard,cat
+```
+
+## Library changes
+### lib:
+two new methods:
+* `lib.Publish`
+* `lib.Update`
+I'll work out how they work while coding.
+
+ListDatasets will have to be adjusted to read from either `repo.PublishedRefs` or `repo.Refs` depending on who's asking.
+
+### p2p:
+* `list_datasets` should now return the list of _published_ datasets.
+* `get_dataset` should only return the latest published version
+
+### repo:
+We'll need to make a new `PublishedRefs` store that is just another-but-separate `RefStore`.
```

Comment:
Alternatively, keep the same RefStore and add a flag to references that says whether they are published or not.

Reply:
oooooh v good idea. that sounds nicer.